tutorials/clinical/10_ieeg_localize.py

```diff
@@ -377,46 +293,51 @@ def plot_overlay(image, compare, title, thresh=None):
-#     for i, ch_coord in enumerate(ch_coords):
-#         # this looks up to a voxel away, it may be marked imperfectly
-#         volume = mne.voxel_neighbors(ch_coord, CT_data, thresh)
-#         for voxel in volume:
-#             if elec_image[voxel] != 0:
-#                 # some voxels ambiguous because the contacts are bridged on
-#                 # the CT so assign the voxel to the nearest contact location
-#                 dist_old = np.sqrt(
-#                     (ch_coords[elec_image[voxel] - 1] - voxel)**2).sum()
-#                 dist_new = np.sqrt((ch_coord - voxel)**2).sum()
-#                 if dist_new < dist_old:
-#                     elec_image[voxel] = i + 1
-#             else:
-#                 elec_image[voxel] = i + 1
-#
-#     # Apply the mapping
-#     warped_elec_image = mapping.transform(elec_image,
-#                                           interpolation='nearest')
-#
-#     # Recover the electrode contact positions as the center of mass
-#     for i in range(ch_coords.shape[0]):
-#         ch_coords[i] = np.array(
-#             np.where(warped_elec_image == i + 1)).mean(axis=1)
-#
-#     # Convert back to surface RAS but to the template surface RAS this time
-#     ch_coords = mne.transforms.apply_trans(
-#         template_brain.header.get_vox2ras_tkr(), ch_coords)
+
+# convert electrode positions from surface RAS to voxels
+ch_coords = mne.transforms.apply_trans(
+    np.linalg.inv(subject_brain.header.get_vox2ras_tkr()), ch_coords)
+# Take channel coordinates and use the CT to transform them
+# into a 3D image where all the voxels over a threshold nearby
+# are labeled with an index
+CT_data = CT_aligned.get_fdata()
+thresh = np.quantile(CT_data, 0.95)
+elec_image = np.zeros(subject_brain.shape, dtype=int)
+for i, ch_coord in enumerate(ch_coords):
+    # this looks up to a voxel away, it may be marked imperfectly
+    volume = mne.voxel_neighbors(ch_coord, CT_data, thresh)
+    for voxel in volume:
+        if elec_image[voxel] != 0:
+            # some voxels ambiguous because the contacts are bridged on
+            # the CT so assign the voxel to the nearest contact location
+            dist_old = np.sqrt(
+                (ch_coords[elec_image[voxel] - 1] - voxel)**2).sum()
+            dist_new = np.sqrt((ch_coord - voxel)**2).sum()
+            if dist_new < dist_old:
+                elec_image[voxel] = i + 1
+        else:
+            elec_image[voxel] = i + 1
+# Apply the mapping
+elec_image = nib.spatialimages.SpatialImage(elec_image, subject_brain.affine)
+del subject_brain, CT_aligned, CT_data  # not used anymore
+
+###############################################################################
+# Warp and plot the result
+_, warped_elec_image = apply_volume_registration(
+    elec_image, template_brain, pre_affine, sdr_morph, interpolation='nearest')
+fig = nilearn.plotting.plot_glass_brain(warped_elec_image)
+fig = fig.axes['x'].ax.figure
```

Comment:
Do you need a different figure, can't you just use the one that was returned?

Reply:
The variable naming here is not great, it doesn't actually return a figure but some custom object. I'll update the var names to make it clearer